src/core/src/energy/global/ewald.rs

```diff
@@ -752,7 +751,7 @@ impl GlobalPotential for SharedEwald {
 
     fn energy(&self, system: &System) -> f64 {
```

Comment:
Is there a reason this has not be done here, or did you just forget ? I can make a PR for this, this way we should be able to completely remove the lock in Ewald :)